sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2516,4 +2516,13 @@ class HiveDDLSuite
       }
     }
   }
+
+  test("SPARK-28443: fail the DDL command if it creates a table with null-type columns") {
+    withTable("t") {
+      val e = intercept[AnalysisException]{
+        sql(s"CREATE TABLE t as SELECT NULL AS c")
```

Comment:
can we also test inner fields?

Reply:
Nested fields are tested in `CatalogSuite`. This UT is just to confirm that cannot using CTAS with `NullType`.